ocaml/typing/typedecl.ml

```diff
@@ -435,24 +463,37 @@ let transl_labels ~new_var_jkind env univars closed lbls =
       lbls in
   lbls, lbls'
 
-let transl_types_gf ~new_var_jkind env univars closed tyl =
-  let mk arg =
+let transl_types_gf ~new_var_jkind ~allow_unboxed
+  env loc univars closed tyl kloc =
+  let mk idx arg =
     let cty = transl_simple_type ~new_var_jkind env ?univars ~closed Mode.Alloc.Const.legacy arg in
     let gf = Typemode.transl_global_flags
       (Jane_syntax.Mode_expr.of_attrs arg.ptyp_attributes |> fst) in
+    check_representable ~why:(Constructor_declaration idx) ~allow_unboxed
+      env loc kloc cty.ctyp_type;
     (cty, gf)
   in
-  let tyl_gfl = List.map mk tyl in
+  let tyl_gfl = List.mapi mk tyl in
```

Comment:
`transl_labels` does the representability check in the second pass, while `transl_types_gf` here does it in the first. Is there a reason for this difference? The change might affect error messages somehow, if there is both an ordinary type error (e.g. `bool int`, because `int` is nullary) and a representability error in different arguments/labels. Probably both functions should do the same thing... and my guess is that doing this in the second pass is better, to prioritize more "normal" type errors.

Reply:
Might be good to add tests that witness the difference.

Reply:
Moved the check to the second pass.